scipy/stats/_continuous_distns.py

```diff
@@ -1516,6 +1516,10 @@ def _sf(self, x, a):
         fac = 0.5*sc.gammainc(a, abs(x))
         return np.where(x > 0, 0.5-fac, 0.5+fac)
 
+    def _entropy(self, a):
+        h = a + np.log(2) + sc.loggamma(a) + (1 - a) * sc.digamma(a)
```

Comment:
Compared to gh-17868, this one is more robust because you're already working with the log of the terms.

However, you'll still run into the second problem here: the `loggamma(a)` term and `a * digamma(a)` term will be huge and of opposite sign. There's a relationship between `loggamma(a)` and `a * digamma(a)` that you can use to cancel out most of them analytically, at least for large `a`. (Taking a look at their series expansions might help to find it.) 

In any case, this is why I suggested in gh-17211 that you open only one of these at a time. This is the same as one of the problems you'll encounter in gh-17868. If you fix the issues in one PR before opening the next, there will be fewer cycles of review in the following PRs.

Reply:
I tested dgamma with ```a = 1e100``` and ```a = 1e-100```, and there was no over/underflow. Unless I'm missing something, the formula should be fine as is.

Reply:
```python3
a = 1
b = np.pi * 1e-17
c = -1
print(a + b + c)
```

Reply:
Even if there is no over/underflow, the result can become imprecise (see the post above). Therefore tests with small values and big values should be added ( e.g. `m=1e-4` and `m=1e20`)